src/AutoPlay.ts

```diff
@@ -1,19 +1,58 @@
 import Flicking, { FlickingEvent, Plugin, Direction } from "@egjs/flicking";
+import { merge } from "./utils";
+
+interface AutoPlayOptions {
+  duration: number;
+  direction: Direction[keyof Direction];
+  stopOnHover: boolean;
+}
+
+const DEFAULT_OPTION: AutoPlayOptions = {
+  duration: 2000,
+  direction: "NEXT",
+  stopOnHover: false,
+};
 
 /**
  * Plugin that allow you to automatically move to the next/previous panel, on a specific time basis
  * @ko 일정 시간마다, 자동으로 다음/이전 패널로 넘어가도록 할 수 있는 플러그인
  * @memberof eg.Flicking.plugins
  */
 class AutoPlay implements Plugin {
+  /* Options */
+  private duration: AutoPlayOptions["duration"];
+  private direction: AutoPlayOptions["direction"];
+  private stopOnHover: AutoPlayOptions["stopOnHover"];
+
+  /* Internal Values */
+  private flicking: Flicking | null = null;
   private timerId = 0;
+
   /**
-   * @param - Time to wait before moving on to the next panel <ko>다음 패널로 움직이기까지 대기 시간</ko>
-   * @param - The direction in which the panel moves. <ko>패널이 움직이는 방향</ko>
+   * @param options Options for the AutoPlay instance.<ko>AutoPlay 옵션</ko>
+   * @param options.duration Time to wait before moving on to the next panel.<ko>다음 패널로 움직이기까지 대기 시간</ko>
+   * @param options.direction The direction in which the panel moves.<ko>패널이 움직이는 방향</ko>
+   * @param options.stopOnHover Whether to stop when mouse hover upon the element.<ko>엘리먼트에 마우스를 올렸을 때 AutoPlay를 정지할지 여부</ko>
    * @example
    * flicking.addPlugins(new eg.Flicking.plugins.AutoPlay(2000, "NEXT"));
    */
-  constructor(private duration: number = 2000, private direction: Direction[keyof Direction] = "NEXT") {}
+  constructor(options: Partial<AutoPlayOptions> = DEFAULT_OPTION, _direction: AutoPlayOptions["direction"] = DEFAULT_OPTION.direction) {
```

Comment:
@WoodNeck 
Shouldn't we use underbar?

Reply:
Oops, maybe my linter was broken. I've fixed it.